tests/src/nodes/nomos.rs

```diff
@@ -58,7 +58,13 @@ impl NomosNode {
     pub async fn spawn(mut config: Config) -> Self {
         // Waku stores the messages in a db file in the current dir, we need a different
         // directory for each node to avoid conflicts
-        let dir = tempfile::tempdir().unwrap();
+        //
+        // NOTE: We could create dirs inside of the OS-default tempdir location.
+        // But, in Windows hosted by Github Actions, it's easier to use the current location
+        // in terms of uploading artifacts to Github,
+        // because source code and system dirs are located on different disks.
+        let dir = tempfile::TempDir::new_in(std::env::current_dir().unwrap()).unwrap();
+
```

Comment:
On Windows hosted by Github Actions,
```
persisting directory C:\Users\RUNNER~1\AppData\Local\Temp\.tmp5A9Xxs -> D:\a\nomos-node\nomos-node\tests\.tmp5A9Xxs
called `Result::unwrap()` on an `Err` value: Os { code: 17, kind: CrossesDevices, message: "The system cannot move the file to a different disk drive." }
```

Reply:
This happened before and the PR resolves it, right?

Reply:
This has never happened before, because we haven't tried to upload artifacts to Github so far ;)

At the begining when I try to implement this PR, I implemented like below:
```rust
let dir = tempfile::tempdir().unwrap();   // using the OS-default
...

fn drop() {
  ...

  // for example,
  std::fs::rename(dir, "$CWD/blahblah"); 
}
```
so that Github actions can upload the artifact `$CWD/blahblah` (also because it's hard to pass the tempdir path in the OS-default location to the Github actions).

But, the `std::fs::rename` failed with the error above: `The system cannot move the file to a different disk drive.`, in Windows. Of course, I could use `copy` instead of `rename`. But, I chose to create a tempdir under the CWD instead of OS-default location.

Reply:
I'll refine the comment in the source code. Thank you!

Reply:
Refined comment: 84965d8b